int32 arm
---
int32 room 